### PR TITLE
Proposed new PR template

### PR DESCRIPTION
**Pull request by shlyaeva**
## Description

Proposal for a much more concise PR template (I dug up my old branch, and trimmed the previous proposal even more).

### CODE REVIEW

**Comment by shlyaeva** [comment]
I can't add Pat Nichols as a reviewer for some reason; guessing he doesn't have access to this repo?

**Review by fmahebert — APPROVED** [review]
I like concise

File: PULL_REQUEST_TEMPLATE.md
Comment (svahl991) [inline PULL_REQUEST_TEMPLATE.md:7]: W could we even maybe remove this section? Github has another place where you can link an associated issue. But it's not as obvious. But I do find that I end up entering the information twice.

File: PULL_REQUEST_TEMPLATE.md
Comment (svahl991) [inline PULL_REQUEST_TEMPLATE.md:11]: I feel the prompt to remember to update documentation might be more appropriate in the "Impact" section than "Dependencies".

File: PULL_REQUEST_TEMPLATE.md
Comment (climbfuji) [reply PULL_REQUEST_TEMPLATE.md:7]: If you use the correct syntax you automatically get the gihub r.h.s populated, and doing it this way is easier to find issues across repositories. I think it's better to keep it

File: PULL_REQUEST_TEMPLATE.md
Comment (climbfuji) [reply PULL_REQUEST_TEMPLATE.md:11]: I agree. A check box or section on documentation updates would be useful.

**Review by eap — APPROVED** [review]
Thank you for making this change; the old template was big enough that half the time I ended up deleting the whole thing and writing my own. I see lots of other people doing the same.

I'm very in favor of this change.

File: PULL_REQUEST_TEMPLATE.md
Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md:7]: I am firmly in support of using "fixes" (or any other keyword with the same functionality) because it automatically closes issues after merging PRs, and I find that very helpful. At least last time I checked the linking of an associated issue did the link but didn't close the issue when the PR was merged. Using "fixes" also creates links in associated issues, so I think it's enough to just use "fixes" (without extra step of linking an issue).

File: PULL_REQUEST_TEMPLATE.md
Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md:11]: @svahl991, @climbfuji: what do you think would be better? A separate _Documentation_ section? Or put in the _Impact_?

Comment (climbfuji) [reply PULL_REQUEST_TEMPLATE.md:11]: Maybe take a look at a PR template from NOAA-GFDL that uses check boxes for a few basic checks: https://github.com/NOAA-GFDL/FMS/pull/1224

File: PULL_REQUEST_TEMPLATE.md
Comment (svahl991) [reply PULL_REQUEST_TEMPLATE.md:7]: Oh I didn't know that "fixes" was a keyword! I always changed the wording for non-bug PR's because I felt "fixes" implied a bug. Maybe that's why it didn't work for me!

File: PULL_REQUEST_TEMPLATE.md
Comment (svahl991) [reply PULL_REQUEST_TEMPLATE.md:11]: > @svahl991, @climbfuji: what do you think would be better? A separate _Documentation_ section? Or put in the _Impact_?

I'd be OK with either. Maybe I lean a little toward putting it in "Impact" just to keep the number of template sections minimal.

File: PULL_REQUEST_TEMPLATE.md
Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md:7]: Some info [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests). We could change it to "closes" or "resolves", they have the same behavior

Comment (svahl991) [reply PULL_REQUEST_TEMPLATE.md:7]: Personally I like "resolves". For what it's worth.

Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md]: I've changed to "resolves".

File: PULL_REQUEST_TEMPLATE.md
Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md]: I've added a checklist as suggested by Dom, I'll re-request all reviews.

**Comment by shlyaeva** [comment]
All: I've added a [very short] section with a checklist (suggested by Dom in one of the comments above) -- what do you think?

**Review by climbfuji — APPROVED** [review]
This is great, thanks very much @shlyaeva !

**Review by climbfuji — APPROVED** [review]
This is great, thanks very much @shlyaeva !

**Review by climbfuji — APPROVED** [review]
This is great, thanks very much @shlyaeva !

**Review by svahl991 — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

File: PULL_REQUEST_TEMPLATE.md
Comment (srherbener) [inline PULL_REQUEST_TEMPLATE.md:11]: In general I like the concise instructions. However, I've gotten tripped up with some subtleties beyond just knowing that this PR goes with another PR (or another set of PRs). For example, if you have a ioda and idoa-data PR set and you happen to merge the ioda PR first, the tests that get kicked off on the new develop branch will use the ioda-data develop branch before the associated ioda-data changes and ctests will fail. Then the test failure shows up on displays such as the status bars in the README.md file and the little checkmark that shows up in github when you show the develop branch (in the "Code" tab. In this case it would be helpful to note that the idoa-data PR needs to be merged before the ioda PR.

I think it might be useful in the Dependencies section to mark the kind of dependency as noted above. Perhaps something like this in the ioda PR
- [ ] merge with ioda-data PR
    merge order: ioda-data, ioda

And for the ioda-data PR
- [ ] merge with ioda PR
    merge order: ioda-data, ioda

Would it make sense to add instructions to specify the merge order in this template?

**Review by srherbener — COMMENTED** [review]
Thanks for putting this together @shlyaeva!

File: PULL_REQUEST_TEMPLATE.md
Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md:11]: I think that the merge order is important to know for repo maintainers (perhaps we can create separate instructions for this to make it clear what the order should be?), but not essential to know or specify for anyone who opens a PR. What do you and others think?

Comment (climbfuji) [reply PULL_REQUEST_TEMPLATE.md:11]: I agree with @shlyaeva, people submitting PRs don't necessarily know this. The template also needs to be independent of the repositories. And the `build group` syntax that will be added as part of the CI pipeline work will cover this aspect to some extend.

Comment (srherbener) [reply PULL_REQUEST_TEMPLATE.md:11]: Good point @shlyaeva. I agree that this information is for repo owners. I'm good with the template as is, and thanks for putting this together.

**Review by srherbener — APPROVED** [review]

**Comment by climbfuji** [comment]
We'll need one more approval from an org owner ... @ytremolet now that the demo is over we can bug you again ;-)

File: PULL_REQUEST_TEMPLATE.md
Comment (ytremolet) [inline PULL_REQUEST_TEMPLATE.md:21]: I would add:
- [ ] I have run the unit tests before creating the PR

**Comment by climbfuji** [comment]
@shlyaeva I have no strong opinion on @ytremolet's suggestion, but I will note that not all repositories have unit tests (either because it's simply not applicable, or because none have been added). Up to you!

File: PULL_REQUEST_TEMPLATE.md
Comment (shlyaeva) [reply PULL_REQUEST_TEMPLATE.md:21]: I've just pushed an update adding this.

**Comment by shlyaeva** [comment]
I think I've resolved all the comments, and this is good to go now. (I don't have the rights to merge)

**Pull request merged**
